File: 14_Longest_Common_Prefix/scan.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int min_len = INT_MAX;
        string ans = "";
        char c;
        
        for (int i = 0; i < strs.size(); i++){
            if (strs[i].length() < min_len){
                min_len = strs[i].length();
            }
        }
        
        for (int i = 0; i < min_len; i++){
            c = strs[0][i];
            for (int j = 1; j < strs.size(); j++){
                if (strs[j][i] != c){
                    return ans;
                }
            }
            ans += c;
        }
        
        return ans;
    }
};


// Time Complexity: O(SN)
// S: The length of the shortest string
// N: The number of strings
// Runtime: 8 ms	Memory: 9.2 MB

File: 14_Longest_Common_Prefix/scan.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        min_len = min([len(strs[i]) for i in range(len(strs))])
        ans = ''
        
        for i in range(min_len):
            c = strs[0][i]
            for j in range(1, len(strs)):
                if strs[j][i] != c:
                    return ans
            ans += c
        
        return ans


# Time Complexity: O(SN)
# S: The length of the shortest string
# N: The number of strings
# Runtime: 56 ms	Memory: 13.9 MB